src/downshift.js

```diff
@@ -639,31 +639,49 @@ class Downshift extends Component {
 
   /////////////////////////////// INPUT
 
-  getInputProps = ({onKeyDown, onBlur, onChange, onInput, ...rest} = {}) => {
+  getInputProps = ({
+    onKeyDown,
+    onBlur,
+    onChange,
+    onInput,
+    onChangeText,
+    ...rest
+  } = {}) => {
     let onChangeKey
+    let eventHandlers = {}
+
     /* istanbul ignore next (preact) */
     if (preval`module.exports = process.env.BUILD_PREACT === 'true'`) {
       onChangeKey = 'onInput'
-      /* istanbul ignore next (react-native) */
-    } else if (
-      preval`module.exports = process.env.BUILD_REACT_NATIVE === 'true'`
-    ) {
-      onChangeKey = 'onChangeText'
     } else {
       onChangeKey = 'onChange'
     }
     const {inputValue, isOpen, highlightedIndex} = this.getState()
-    const eventHandlers = rest.disabled
-      ? {}
-      : {
-          [onChangeKey]: callAllEventHandlers(
-            onChange,
-            onInput,
-            this.input_handleChange,
-          ),
-          onKeyDown: callAllEventHandlers(onKeyDown, this.input_handleKeyDown),
-          onBlur: callAllEventHandlers(onBlur, this.input_handleBlur),
-        }
+
+    if (!rest.disabled) {
```

Comment:
I don't love how this forking logic came out, so I'd love some feedback here.

Reply:
Actually, I don't think it's all that bad :+1: